pkg/apis/serving/v1/stagepodautoscaler_types.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2023 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"knative.dev/pkg/apis"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	"knative.dev/pkg/kmeta"
+)
+
+// +genclient
+// +genreconciler
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// StagePodAutoscaler is a Knative abstraction that encapsulates the interface.
+type StagePodAutoscaler struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec holds the desired state of the StagePodAutoscaler (from the client).
+	// +optional
+	Spec StagePodAutoscalerSpec `json:"spec,omitempty"`
+
+	// Status holds the desired state of the StagePodAutoscaler (from the client).
+	// +optional
+	Status StagePodAutoscalerStatus `json:"status,omitempty"`
+}
+
+const (
+	SPAReady apis.ConditionType = "PodAutoscalerStageReady"
+)
+
+// StagePodAutoscalerStatus communicates the observed state of the StagePodAutoscaler (from the controller).
+type StagePodAutoscalerStatus struct {
+	duckv1.Status `json:",inline"`
+
+	// DesiredScale shows the current desired number of replicas for the revision.
+	DesiredScale *int32 `json:"desiredScale,omitempty"`
+
+	// ActualScale shows the actual number of replicas for the revision.
+	ActualScale *int32 `json:"actualScale,omitempty"`
```

Comment:
These fields already exist on KPA autoscaler, can we update KPA status object instead?

Reply:
Knative already create too many child custom resources , let's try not creating new ones unless we really need to.

Reply:
This is the good point, but I really need this ActualScale in the status of this stagepodautoscaler.
This reason is as below:
```
This change in podautoscaler is only able to kick-off the reconcile loop of the revision. However, I need any change in podautoscaler to kick off the reconcile loop of the service orchestrator.
So I have to use podautoscaler to kick off the reconcile loop of stagepodautoscaler, since they share the same name. And with the change of `ActualScale ` in the stagepodautoscaler to kick off the reconcile loop of the service orchestrator.
```